Homework4/DiagonalMatrix/Extensions/DiagonalMatrixExtension.cs

```diff
@@ -0,0 +1,42 @@
+﻿
+namespace DiagonalMatrix
+{
+    public static class DiagonalMatrixExtension
+    {
+        public delegate T MatrixAddition<T>(T element1, T element2);
+
+        public static DiagonalMatrix<T> AddMatrices<T>(this DiagonalMatrix<T> matrix1, DiagonalMatrix<T> matrix2, 
+            MatrixAddition<T> addition)
+        {
+            T[] newElementsAfterAddition;
+
+            if (matrix1.Size == matrix2.Size) // For the matrices with the same dimension
```

Comment:
You can drop the lines in if {...} and keep only else {...} 

Consider the pseudo code ( else block, line 25, **>** replaced to **>=**)

```
var biggerMatrix = matrix1.Size >= matrix2.Size ? matrix1 : matrix2;
var smallerMatrix = matrix1.Size < matrix2.Size ? matrix1 : matrix2;
```
With that update, for cycle below will stop immediately and the case of equal matrixes will be handler properly. 
`for (int i = smallerMatrix.Size; i < biggerMatrix.Size; i++)`